3rdparty/fastcv/src/fastcv_hal_imgproc.cpp

```diff
@@ -52,6 +52,97 @@ int fastcv_hal_medianBlur(
     CV_HAL_RETURN(status, hal_medianBlur);
 }
 
+class FcvSobelLoop_Invoker : public cv::ParallelLoopBody
+{
+    public:
+
+    FcvSobelLoop_Invoker(const uchar* _src_data, size_t _src_step, uchar* _dst_data, size_t _dst_step, int _width,
+        int _height, int _src_depth, int _dst_depth, int _dx, int _dy, int _ksize, fcvBorderType _fcvBorder, int _fcvBorderValue) :
+        cv::ParallelLoopBody(), src_data(_src_data), src_step(_src_step), dst_data(_dst_data), dst_step(_dst_step), width(_width),
+        height(_height), src_depth(_src_depth), dst_depth(_dst_depth), dx(_dx), dy(_dy), ksize(_ksize), fcvBorder(_fcvBorder),
+        fcvBorderValue(_fcvBorderValue)
+    {
+        half_ksize  = ksize/2;
+        fcvFuncType = FCV_MAKETYPE(ksize,src_depth);
+    }
+
+    virtual void operator()(const cv::Range& range) const CV_OVERRIDE
+    {
+        int topLines    = 0;
+        int rangeHeight = range.end-range.start;
+
+        if(range.start > 0)
+        {
+            topLines    += half_ksize;
+            rangeHeight += half_ksize;
+        }
+
+        if(range.end < height)
+        {
+            rangeHeight += half_ksize;
+        }
+
+        const uchar* src = src_data + (range.start-topLines)*src_step;
+
+        std::vector<uchar> dst(dst_step * rangeHeight);
+        int16_t *dxBuffer = nullptr, *dyBuffer = nullptr;
+
+        if ((dx == 1) && (dy == 0))
+        {
+            dxBuffer = (int16_t*)dst.data();
+        }
+        else if ((dx == 0) && (dy == 1))
+        {
+            dyBuffer = (int16_t*)dst.data();
+        }
+
+        switch (fcvFuncType)
+        {
+            case FCV_MAKETYPE(3,CV_8U):
+            {
+                fcvFilterSobel3x3u8s16(src, width, rangeHeight, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            case FCV_MAKETYPE(5,CV_8U):
+            {
+                fcvFilterSobel5x5u8s16(src, width, rangeHeight, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            case FCV_MAKETYPE(7,CV_8U):
+            {
+                fcvFilterSobel7x7u8s16(src, width, rangeHeight, src_step, dxBuffer, dyBuffer, dst_step, fcvBorder, 0);
+                break;
+            }
+            default:
```

Comment:
I propose to add at least CV_Assert here. Just for the case, if invoker code will be changed incorrectly in future.

Reply:
which param do you think we should add assert? or I can raise some error messages as it already in parallel we can't return CV_HAL_ERROR_NOT_IMPLEMENTED

Reply:
You can use just `CV_Error()` statement then. The branch should not be reachable in normal configuration.